src/views/dependencyDataProvider.ts

```diff
@@ -67,6 +68,37 @@ export class DependencyDataProvider implements TreeDataProvider<ExplorerNode> {
         return element.getParent();
     }
 
+    public async getRootNodeByData(nodeData: INodeData): Promise<DataNode> {
+        // Server only return project nodes, so use get root projects function
```

Comment:
As said, in order to get the data, creating the view is not an acceptable solution here.  DON't mix the model with your view. You either either directly get the data from the server, or patch the Workspsace data to the path side.